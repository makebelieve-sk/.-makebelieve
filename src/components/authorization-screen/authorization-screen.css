.login {
    display: flex;
    align-items: center;
    flex-direction: column;
    flex-wrap: wrap;
    width: 100%;
    height: 100%; 
}
.login_logo {
    margin: 0;
    margin-top: 220px; 
}
.login_title {
    margin: 0;
    font-family: "Fira Sans", "Arial", sans-serif;
    font-style: italic;
    font-weight: 300;
    font-size: 33px;
    color: #230d1a;
    text-align: center;
    margin-top: 10px; 
}
.login_total {
    margin: 0;
    font-family: "Fira Sans", "Arial", sans-serif;
    font-style: italic;
    font-weight: 300;
    font-size: 33px;
    color: #230d1a;
    text-align: center;
    font-size: 20px;
    margin-top: 30px;
    width: 500px; 
}
.login_total--fail {
    margin-top: 170px; 
}
.login_text {
    margin: 0;
    font-family: "Fira Sans", "Arial", sans-serif;
    font-style: italic;
    font-weight: 300;
    font-size: 33px;
    color: #230d1a;
    text-align: center;
    font-size: 20px;
    margin-top: 100px;
    width: 500px; 
}
.login_form {
    width: 450px;
    margin-top: 20px;
    display: flex;
    flex-wrap: wrap;
    justify-content: center; 
}
.login_field {
    position: relative;
    flex-basis: 50%;
    width: 50%;
    margin: 0; 
}
/* .login_field:last-of-type {
    text-align: right; 
} */
.login_label {
    font-family: "Fira Sans", "Arial", sans-serif;
    font-style: normal;
    font-weight: 500;
    margin-right: 5px; 
}
.login_input {
    padding: 5px 10px;
    width: 120px;
    background-color: #f0eed5;
    border: 3px solid #230d1a;
    border-radius: 15px; 
}
.login_input:hover {
    border-color: #ff9749; 
}
.login_input--error {
    border-color: #f00000; 
}
.login_input--error:hover {
    border-color: #f00000; 
}
.login_error {
    font-family: "Fira Sans", "Arial", sans-serif;
    font-style: normal;
    font-weight: 300;
    font-size: 16px;
    display: none;
    position: absolute;
    bottom: -20px;
    right: 10px;
    font-size: 14px;
    color: #f00000; 
}
.login_input--error + .login_error {
    display: inline; 
}
.login_button {
    margin-top: 10px; 
    margin-bottom: 10px;
}
.login_button:hover {
    cursor: pointer;
}
.login_replay {
    position: relative;
    margin-top: 20px;
    padding: 0;
    padding-right: 40px;
    color: #ff9749;
    font-size: 30px;
    border: none;
    background: transparent; 
}
.login_replay::after {
    position: absolute;
    content: "";
    width: 27px;
    height: 25px;
    top: 50%;
    right: 0;
    /* background-image: url("../img/sprite.png"); */
    background-position: -5px -357px;
    -webkit-transform: translateY(-50%);
            transform: translateY(-50%); 
}
.login_replay:hover, .login_replay:focus {
    color: #f00000; 
}
.login_replay:hover::after, .login_replay:focus::after {
    background-position: -5px -392px; 
}