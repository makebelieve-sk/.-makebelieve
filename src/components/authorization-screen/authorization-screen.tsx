import React from 'react';
import { Link } from 'react-router-dom';

import './authorization-screen.css';

export const AuthorizationScreen = () => {
    return (
      <section className="login">
        <div className="login_logo">
          <img src="./img/melody-logo.png" alt="Угадай мелодию" width="186" height="83" />
        </div>
        <h2 className="login_title">Необходима авторизация</h2>
        <p className="login_text">Представтесь!</p>
        <form className="login_form" action="">
          <p className="login_field">
            <label className="login_label" htmlFor="name">Логин</label>
            <input className="login_input" type="text" name="name" id="name" />
          </p>
          <p className="login_field">
            <label className="login_label" htmlFor="password">Пароль</label>
            <input className="login_input" type="text" name="password" id="password" />
            <span className="login_error">Неверный пароль</span>
          </p>
          <a href="#"><button className="login_button button" type="submit">Зарегистрироваться</button></a>
        </form>
      </section>
    );
};