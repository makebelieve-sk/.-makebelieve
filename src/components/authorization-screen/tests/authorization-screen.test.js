import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { AuthorizationScreen } from '../authorization-screen';

Enzyme.configure({ adapter: new Adapter() });

const setUp = () => shallow(<AuthorizationScreen />);

describe(`AuthorizationScreen component`, () => {
    let component;

    beforeEach(() => {
        component = setUp();
    });

    it(`should contain section wrapper`, () => {
        const wrapper = component.find('.login');
        expect(wrapper.length).toBe(1);
    });

    it(`AuthorizationScreen correctly renders after relaunch`, () => {
        const tree = renderer  
            .create(<AuthorizationScreen />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
