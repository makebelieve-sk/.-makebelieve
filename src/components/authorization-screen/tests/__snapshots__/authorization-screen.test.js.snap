// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorizationScreen component AuthorizationScreen correctly renders after relaunch 1`] = `
<section
  className="login"
>
  <div
    className="login_logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="./img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="login_title"
  >
    Необходима авторизация
  </h2>
  <p
    className="login_text"
  >
    Представтесь!
  </p>
  <form
    action=""
    className="login_form"
  >
    <p
      className="login_field"
    >
      <label
        className="login_label"
        htmlFor="name"
      >
        Логин
      </label>
      <input
        className="login_input"
        id="name"
        name="name"
        type="text"
      />
    </p>
    <p
      className="login_field"
    >
      <label
        className="login_label"
        htmlFor="password"
      >
        Пароль
      </label>
      <input
        className="login_input"
        id="password"
        name="password"
        type="text"
      />
      <span
        className="login_error"
      >
        Неверный пароль
      </span>
    </p>
    <a
      href="#"
    >
      <button
        className="login_button button"
        type="submit"
      >
        Зарегистрироваться
      </button>
    </a>
  </form>
</section>
`;
